lib/beaker/host/freebsd/pkg.rb

```diff
@@ -1,13 +1,28 @@
 module FreeBSD::Pkg
   include Beaker::CommandFactory
 
-  def install_package(name, cmdline_args = nil, opts = {})
-    cmdline_args ||= '-y'
-    execute("pkg install #{cmdline_args} #{name}", opts) { |result| result }
+  def shell_check_pkgng
+    # See manpage of pkg(8), the paragraph devoted to -N flag
+    # https://www.freebsd.org/cgi/man.cgi?query=pkg
+    'TMPDIR=/dev/null ASSUME_ALWAYS_YES=1 PACKAGESITE=file:///nonexist ' \
+    'pkg info -x "pkg(-devel)?\\$" > /dev/null 2>&1'
   end
 
-  def check_for_package(name, opts = {})
-    execute("pkg info #{name}", opts) { |result| result }
+  def shell_ifelse(cond, stmt_t, stmt_f)
+    "if #{cond}; then #{stmt_t}; else #{stmt_f}; fi"
   end
 
+  def install_package(package, cmdline_args = nil, opts = {})
+    cmd = shell_ifelse shell_check_pkgng,
+                       "pkg install #{cmdline_args || '-y'} #{package}",
+                       "pkg_add #{cmdline_args || '-r'} #{package}"
+    execute("/bin/sh -c '#{cmd}'", opts) { |result| result }
```

Comment:
Well, by default FreeBSD uses `csh` shell, which has different syntax than `sh`. This line "makes the code to work" but it eventually changes shell. If there are some environment variables set only for the `csh` shell and are not set for `sh`, then some problems may appear here.